sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala

```diff
@@ -201,35 +216,33 @@ class SerializeConcatHostBuffersDeserializeBatch(
     }
   }
 
-  def numRows: Int = {
-    if (batchInternal != null) {
-      batchInternal.numRows()
-    } else {
-      headers.map(_.getNumRows).sum
-    }
-  }
+  def numRows: Int = Option(batchInternal)
+    .map(_.fold(_.numRows(), _.numRows()))
+    .getOrElse(headers.map(_.getNumRows).sum)
 
-  def dataSize: Long = {
-    if (batchInternal != null) {
-      val bases = GpuColumnVector.extractBases(batchInternal).map(_.copyToHost())
+  def dataSize: Long = Option(batchInternal)
+    .map { maybeSpillable =>
+      val batch = maybeSpillable.fold(identity, _.getColumnarBatch())
```

Comment:
I thought there was a data size API for SpillableColumnarBatch?  Also don't we have to close the batch if we got it from SerializableColumnarBatch?